src/server/qgsserverparameters.cpp

```diff
@@ -200,6 +206,139 @@ QgsRectangle QgsServerParameterDefinition::toRectangle( bool &ok ) const
   return extent;
 }
 
+QString QgsServerParameterDefinition::loadUrl( bool &ok ) const
+{
+  ok = true;
+
+  // Get URL
+  QUrl url = toUrl( ok );
+  if ( !ok )
+  {
+    return QString();
+  }
+
+  // Do the request
+  QNetworkReply *reply = nullptr;
+  // The following code blocks until the file is downloaded...
```

Comment:
Is there some kind of timeout mechanism in this case (to avoid an infinite loop)?

Reply:
Which duration ?

Reply:
@pblottiere I have added a timeout mecahnism.